homeassistant/helpers/entity.py

```diff
@@ -597,7 +597,7 @@ def use_device_name(self) -> bool:
         ) and name_translation_key in self.platform_data.platform_translations:
             return False
         if hasattr(self, "entity_description"):
-            return not self.entity_description.name
+            return self.entity_description.name not in (UNDEFINED, None)
```

Comment:
UNDEFINED is a sentinel that shouldn't end up here.

How did it get here in the first place?

../Frenck

Reply:
Thanks for the review. The reason for this change is that `EntityDescription` 's name attribute defaults to `UNDEFINED`.

While it ideally might be handled earlier, I am not confident there is a guarantee within the core that this `UNDEFINED` value is always validated or resolved before reaching this helper.

This means an integration can still pass an `EntityDescription` with the default `UNDEFINED` name (intentionally or not, may be they intend to forward value is None), and this helper needs to be robust against that case.

I'll keep this PR as a draft for now. If this isn't the right place for this check, could you point me to where you think this validation should ideally occur?

Thank you

Reply:
@frenck Hi, kindly ping to you for above message.